eng-wasm/derive/examples/trait-impl-rename.rs

```diff
@@ -0,0 +1,29 @@
+#![no_std]
+
+use eng_wasm_derive::pub_interface;
+
+struct MyContractImplementation;
+
+#[pub_interface(MyContractImplementation)]
+trait MyContract {
+    /// constructor
+    fn construct(_x: u32);
+
+    /// secret contract method
+    fn expand(input: u32) -> u64;
+}
+
+impl MyContract for MyContractImplementation {
```

Comment:
I would maybe add `impl MyContract` for another structure in addition, just to emphasize the point of the argument to `pub_interface`. But it is up to you.